plot_features.py

```diff
@@ -132,40 +78,66 @@ def plot_features_file(featurefile, mapInfo): #{{{
+    linestyle = ['-', '--', '-.', ':']
 
     feature_num = 0
 
+    bounds = None
+
     for feature in featuredat['features']:
-        polytype = feature['geometry']['type']
-        coords = feature['geometry']['coordinates']
-        feature = feature['properties']['name']
-        print '  feature: %s'%feature
+        geomtype = feature['geometry']['type']
+        shape = shapely.geometry.shape(feature['geometry'])
+        featurename = feature['properties']['name']
+        print '  feature: %s'%featurename
+
+        refProjection = cartopy.crs.PlateCarree()
+
+        color = colors[feature_num % len(colors)]
+        linestyle = linestyle[feature_num % len(linestyle)]
+        marker = markers[feature_num % len(markers)]
+
+        props = {'linewidth':2.0, 'edgecolor':color}
+        if geomtype in ['Polygon','MultiPolygon']:
+            props['alpha'] = 0.4
+            props['facecolor'] = color
+        if geomtype in ['LineString','MultiLineString']:
+            props['linestyle'] = linestyle
+            props['facecolor'] = 'none'
+        if geomtype in ['Point']:
+            props['marker'] = marker
+
+        if bounds is None:
+            bounds = list(shape.bounds)
+        else:
+            # expand the bounding box
+            bounds[0] = min(bounds[0],shape.bounds[0])
+            bounds[1] = min(bounds[1],shape.bounds[1])
+            bounds[2] = max(bounds[2],shape.bounds[2])
+            bounds[3] = max(bounds[3],shape.bounds[3])
 
-        color_num = feature_num % len(colors)
-        marker_num = feature_num % len(markers)
+        for mapType in mapInfo:
+            (ax, projection, plotFileName, fig) = mapInfo[mapType]
 
-        try:
-            if polytype == 'MultiPolygon':
-                for poly in coords:
-                    for shape in poly:
-                        plot_poly(mapInfo,shape,colors[color_num])
-            elif polytype == 'Polygon' or polytype == 'MultiLineString':
-                for poly in coords:
-                    plot_poly(mapInfo,poly,colors[color_num])
-            elif polytype == 'LineString':
-                plot_poly(mapInfo,coords,colors[color_num],filled=False)
-            elif polytype == 'Point':
-                plot_point(mapInfo,coords,markers[marker_num],colors[color_num])
-            else:
-                assert False, 'Geometry %s not known.'%(polytype)
-        except:
-            print 'Error plotting %s'%(feature)
+            ax.add_geometries((shape,), crs=refProjection, **props)
 
         feature_num = feature_num + 1
 
+    box = shapely.geometry.box(bounds[0],bounds[1],bounds[2],bounds[3])
+
+
     for mapType in mapInfo:
         (ax, projection, plotFileName, fig) = mapInfo[mapType]
         print 'saving ' + plotFileName
+
+        boxProjected = projection.project_geometry(box, src_crs=refProjection)
+        try:
+            x1, y1, x2, y2 = boxProjected.bounds
+            ax.set_xlim([x1, x2])
+            ax.set_ylim([y1, y2])
```

Comment:
If you wanted to remove a line of code I think you could do something like

```python
ax.set_xlim(*boxProjected.bounds[0,2])
ax.set_ylim(*boxProjected.bounds[1,3])
```

However, this makes things a little less clear and you may not want to make this change.